corehq/apps/users/util.py

```diff
@@ -314,26 +314,25 @@ def _last_sync_needs_update(last_sync, sync_datetime):
     return False
 
 
-def log_user_role_update(domain, user, by_user, updated_via):
+def log_user_role_update(domain, user_role, user, by_user, updated_via):
     """
-    :param domain: domain name
+    :param domain: domain that initiated the change
+    :param user_role: user's new role
     :param user: couch user that got updated
-    :param by_user: django/couch user that made the update
+    :param by_user: couch user that made the update
     :param updated_via: web/bulk_importer
     """
-    user_role = user.get_role(domain)
-    message = "role: None"
+    changes = {'role': None}
+    message = ''
     if user_role:
-        if user_role.get_qualified_id() == 'admin':
-            message = f"role: {user_role.name}"
-        else:
-            message = f"role: {user_role.name}[{user_role.get_id}]"
-    message += f", updated_via: {updated_via}"
-    log_model_change(by_user, user.get_django_user(), message=message)
+        changes['role'] = user_role.get_qualified_id()
+        message = f"role: {user_role.name}"
+    log_user_change(domain, user, by_user, changed_via=updated_via, message=message, fields_changed=changes)
 
 
 def log_user_change(domain, couch_user, changed_by_user, changed_via=None,
-                    message=None, fields_changed=None, action=ModelAction.UPDATE):
+                    message=None, fields_changed=None, action=ModelAction.UPDATE,
+                    can_skip_domain=False):
```

Comment:
I assume `can_skip_domain` will get renamed when you merge these PRs together.

Reply:
Yes!

Reply:
https://github.com/dimagi/commcare-hq/pull/29890/commits/dc47bf5d4ded6d7d169d535c53ae04d0b7abc100

and

https://github.com/dimagi/commcare-hq/pull/29890/commits/84a87db6900230a1ab1776715101adbf36ef7c5f